pkg/statistics/handle/autoanalyze/priorityqueue/dynamic_partitioned_table_analysis_job.go

```diff
@@ -35,50 +36,54 @@ const (
 )
 
 // DynamicPartitionedTableAnalysisJob is a TableAnalysisJob for analyzing dynamic pruned partitioned table.
+//
+//nolint:fieldalignment
 type DynamicPartitionedTableAnalysisJob struct {
-	// Only set when partitions's indexes need to be analyzed.
-	// It looks like: {"indexName": ["partitionName1", "partitionName2"]}
-	// This is only for newly added indexes.
-	// The reason why we need to record the partition names is that we need to analyze partitions in batch mode
-	// and we don't want to analyze the same partition multiple times.
-	// For example, the user may analyze some partitions manually, and we don't want to analyze them again.
-	PartitionIndexes map[string][]string
+	successHook SuccessJobHook
+	failureHook FailureJobHook
 
-	successHook JobHook
-	failureHook JobHook
+	SchemaName        pmodel.CIStr
+	GlobalTableID     int64
+	PartitionIDs      map[int64]struct{}
+	PartitionIndexIDs map[int64][]int64
 
-	TableSchema     string
-	GlobalTableName string
-	// This will analyze all indexes and columns of the specified partitions.
-	Partitions []string
 	// Some indicators to help us decide whether we need to analyze this table.
 	Indicators
-	GlobalTableID int64
-
 	// Analyze table with this version of statistics.
 	TableStatsVer int
 	// Weight is used to calculate the priority of the job.
 	Weight float64
+
+	// Lazy initialized.
```

Comment:
Maybe we should put these fields into a substruct. But the PR is too big. I will do it later.